### PR TITLE
Adding list support

### PR DESCRIPTION
**Pull request by epipho**
Adds List support to all backends + crypt app. Nearly complete, needs more testing + unit test written for config.

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -98,6 +105,39 @@ func (c standardConfigManager) Get(key string) ([]byte, error) {
...
+// convenience function, no additional value provided over
+// `etcdctl`
+func (c standardConfigManager) List(key string) (KVPairs, error) {
+	list, err := c.store.List(key)
+	retList := make(KVPairs, len(list))
+	if err != nil {
+		return nil, err
+	}
+	for i, kv := range list {
+		retList[i].Key = kv.Key
```
Comment (epipho) [inline config/config.go]: This is to prevent leaking the backend.KVPairs type out of the library, is there a nicer way to do this?

**Comment by epipho** [comment]
Is there a reason the crypt app uses the backend directly instead of creating and using a ConfigManager? Seems like it would make for nicer code and provide a better example app.

**Comment by epipho** [comment]
Tested with both etcd and consul backends.

File: config/config.go (hunk 1 above)
Comment (bketelsen) [inline config/config.go]: Not that I know

**Comment by bketelsen** [comment]
Thanks for the PR.

**Pull request merged**